### PR TITLE
Add test cases covering RTR ASPA Announcements + Withdrawals

### PR DESCRIPTION
**Pull request by carl-tud**
This PR aims to introduce new test cases that verify the RTR update/undo functionality works as expected.

### CODE REVIEW

**Comment by carl-tud** [comment]
Looks good overall, will check later

File: tests/test_aspa.c [hunk 1]
```diff
@@ -7,351 +7,412 @@
...
 #include <string.h>
 
-/*
-static void update_cb(struct aspa_table *t __attribute__((unused)), const struct aspa_record rec, const struct rtr_socket *socket, const bool added)
+char *data;
+size_t data_size = 0;
+size_t data_index = 0;
+
+#define c16(X) lrtr_convert_short(TO_HOST_HOST_BYTE_ORDER, X)
+#define c32(X) lrtr_convert_long(TO_HOST_HOST_BYTE_ORDER, X)
```
Comment (carl-tud) [inline tests/test_aspa.c]: This should should be named something more descriptive

Comment (moversity) [reply tests/test_aspa.c]: please give suggestions. i wanted to keep it very short: convert 16 bit / convert 32 bit

**Comment by moversity** [comment]
i changed the base branch to aspa-fixes to not focus on the technical fixes which these tests require to function

File: tests/test_aspa.c (hunk 1 above)
Comment (carl-tud) [reply tests/test_aspa.c]: maybe we could call it `HOST_BYTES`

**Review by tanneberger — APPROVED** [review]

**Comment by moversity** [comment]
I believe this PR can be closed because all relevant changes have been merged to master

**Pull request merged**